sklearn/utils/estimator_checks.py

```diff
@@ -687,7 +691,11 @@ def check_fit_score_takes_y(name, Estimator):
 @ignore_warnings
 def check_estimators_dtypes(name, Estimator):
     rnd = np.random.RandomState(0)
-    X_train_32 = 3 * rnd.uniform(size=(20, 5)).astype(np.float32)
+    if name in ["BoxCoxTransformer", "SpectralCoclustering",
```

Comment:
I'm not very familiar with the algorithm but that seems strange to me. Have you understood why this is happening?